### PR TITLE
Update #545 - updates typeDefs and imports for Challenge and Submission

### PR DESCRIPTION
**Pull request by kondanna**
This would have to be merged at the same time as the c0d3-cli PR, or submissions won't work.

I tested createSubmission with the CLI and it was successful for all lessons 1-7 (doesn't work on 8 or 9). The submissions also show on the curriculum page and can be reviewed and rejected.

Only the Id on Lesson changed to Int ,  the rest will be updated on a future PR. 

I realize this was a lot of files, but submissions depend on challenge and lesson and I figured I'd switch some imports from @types to graphql generated ones.

### CODE REVIEW

**Comment by kondanna** [comment]
hi, testing discord github webhook

**Comment by kondanna** [comment]
testing discord github webhook

**Comment by kondanna** [comment]
testing discord webhook

**Comment by kondanna** [comment]
testing discord webhook

**Comment by kondanna** [comment]
testing discord webhook

File: helpers/controllers/submissionController.ts [hunk 1]
```diff
@@ -9,14 +9,14 @@ import _ from 'lodash'
 const { User, Submission, Challenge, Lesson } = db
 
 type ArgsCreateSubmission = {
-  lessonId: string
+  lessonId: number
   cliToken: string
   diff: string
-  challengeId: string
+  challengeId: number
 }
```
Comment (ggwadera) [inline helpers/controllers/submissionController.ts:16]: The argument types are auto-generated as well. You remove the `ArgsCreateSubmission` and `ArgsGetSubmissions` types and do the following:

```ts
import {
  MutationCreateSubmissionArgs,
  QuerySubmissionsArgs
} from '../../graphql'

export const createSubmission = async (
  _parent: void,
  args: MutationCreateSubmissionArgs
) => { // ... }

export const submissions = async (
  _parent: void,
  arg: QuerySubmissionsArgs,
  ctx: Context
) => { // ... }
```

File: pages/curriculum/[lesson].tsx [hunk 2]
```diff
@@ -6,33 +6,51 @@ import LessonTitleCard from '../../components/LessonTitleCard'
...
   QueryDataProps
 } from '../../containers/withQueryLoader'
 import _ from 'lodash'
 
-const Challenges: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
+const Challenges: React.FC<QueryDataProps<GetAppQuery>> = ({ queryData }) => {
   const { lessons, session, alerts } = queryData
   const router = useRouter()
-  const currentlessonId = router.query.lesson as string
+  const currentlessonId = Number(router.query.lesson) as number
```
Comment (ggwadera) [inline pages/curriculum/[lesson].tsx:24]: This is like if you said `Take this number as a number`, pretty redundant right? The compiler is smart enough to figure out that it's a number haha.

```suggestion
  const currentlessonId = Number(router.query.lesson)
```

File: pages/curriculum/[lesson].tsx [hunk 3]
```diff
@@ -6,33 +6,51 @@ import LessonTitleCard from '../../components/LessonTitleCard'
...
   if (!lessons || !alerts) {
     return <Error code={StatusCode.INTERNAL_SERVER_ERROR} message="Bad data" />
   }
   const currentLesson: Lesson | undefined = lessons.find(
-    (lesson: Lesson) => lesson.id === currentlessonId
-  )
+    lesson => lesson.id === currentlessonId
+  ) as Lesson
```
Comment (ggwadera) [inline pages/curriculum/[lesson].tsx:30]: If you remove the type assertions this works without issues, the compiler can infer the type automatically.

```ts
const currentLesson = lessons.find(lesson => lesson.id === currentlessonId)
```

File: pages/review/[lesson].tsx [hunk 4]
```diff
@@ -7,34 +7,32 @@ import LessonTitleCard from '../../components/LessonTitleCard'
...
-  submissions: SubmissionData[]
+  submissions: Submission[]
 }
 
 const SubmissionDisplay: React.FC<SubmissionDisplayProps> = ({
   submissions
 }) => (
   <div className="submissions-container container p-0">
-    {submissions.map((submission: SubmissionData) => (
+    {submissions.map((submission: Submission) => (
```
Comment (ggwadera) [inline pages/review/[lesson].tsx:25]: Redundancy.

```suggestion
    {submissions.map((submission) => (
```

File: pages/review/[lesson].tsx [hunk 5]
```diff
@@ -7,34 +7,32 @@ import LessonTitleCard from '../../components/LessonTitleCard'
...
   </div>
 )
 
-const Review: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
+const Review: React.FC<QueryDataProps<GetAppQuery>> = ({ queryData }) => {
   const { lessons, session } = queryData
   const router = useRouter()
-  const currentlessonId = router.query.lesson as string
+  const currentlessonId = Number(router.query.lesson)
+  console.log(currentlessonId)
```
Comment (ggwadera) [inline pages/review/[lesson].tsx:35]: Looks like you forgot a console.log here.

File: pages/review/[lesson].tsx [hunk 6]
```diff
@@ -46,22 +44,22 @@ const Review: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
     return <LoadingSpinner />
   }
   const currentLesson: Lesson | undefined = lessons.find(
-    (lesson: Lesson) => lesson.id === currentlessonId
-  )
+    lesson => lesson.id === currentlessonId
+  ) as Lesson
   if (!currentLesson) {
```
Comment (ggwadera) [inline pages/review/[lesson].tsx:49]: Redundancy. Same case as the other one, type can be inferred.

```ts
const currentLesson = lessons.find(lesson => lesson.id === currentlessonId)
```

**Comment by kondanna** [comment]
wow, thanks for these!

File: helpers/controllers/submissionController.ts [hunk 7]
```diff
@@ -11,14 +11,14 @@ import { prisma } from '../../prisma'
...
   diff: string
   challengeId: string
 }
 
 type ArgsGetSubmissions = {
-  lessonId: string
+  lessonId: number
 }
 
 export const createSubmission = async (
```
Comment (ggwadera) [inline helpers/controllers/submissionController.ts:24]: Use the types I mentioned here https://github.com/garageScript/c0d3-app/pull/629#discussion_r612877071

File: pages/curriculum.tsx [hunk 8]
```diff
@@ -87,9 +87,9 @@ export const Curriculum: React.FC<Props> = ({ lessons, alerts }) => {
   React.useEffect(() => {
     if (data && data.session) {
       setState({
-        session: data.session,
-        progress: calculateProgress(data.session, lessons),
-        current: calculateCurrent(data.session, lessons)
+        session: data.session as Session,
+        progress: calculateProgress(data.session as Session, lessons),
+        current: calculateCurrent(data.session as Session, lessons)
```
Comment (ggwadera) [inline pages/curriculum.tsx:92]: I fixed these types on #725, assertion is not needed.

```suggestion
        session: data.session,
        progress: calculateProgress(data.session, lessons),
        current: calculateCurrent(data.session, lessons)
```

**Review by songz — APPROVED** [review]

**Pull request merged**